### PR TITLE
DOC: special: add full docstring for `expi`

### PR DESCRIPTION
**Pull request by person142**
#### Reference issue

None

#### What does this implement/fix?

A full docstring for `special.expi` is added.

#### Additional information

None.

### CODE REVIEW

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -2601,16 +2601,90 @@ def add_newdoc(name, doc):
...
+    The SciPy implementation returns the real variant for complex
+    values on the branch cut.
 
-        integral(exp(t)/t, t=-inf..x)
+    >>> sc.expi(complex(-1, 0.0))
+    (-0.21938393439552062-0j)
+    >>> sc.expi(complex(-1, -0.0))
+    (-0.21938393439552062-0j)
 
-    See `expn` for a different exponential integral.
```
Comment (ev-br) [inline scipy/special/add_newdocs.py]: Nit, optional: worth keeping the seealso expn?

**Review by ev-br — APPROVED** [review]
LGTM modulo an optional suggestion.

File: scipy/special/add_newdocs.py (hunk 1 above)
Comment (person142) [reply scipy/special/add_newdocs.py]: Done.

**Comment by ev-br** [comment]
Thanks Josh

**Pull request merged**